version: "3.9"

services:
  mysql:
    container_name: mysql
    image: mysql:8.0
    platform: linux/amd64
    environment:
      MYSQL_ROOT_PASSWORD: "P@55W0rD"
    ports:
      - "3366:3306"
    healthcheck:
      test: [ "CMD", "mysqladmin" ,"ping", "-h", "localhost" ]
      interval: 5s
      timeout: 10s
      retries: 10
    volumes:
      - ./docker/mysql/data:/var/lib/mysql
      - ./docker/mysql/script:/opt/scripts
      - ./docker/mysql/script/setup.sql:/docker-entrypoint-initdb.d/1.sql
    command: --default-authentication-plugin=mysql_native_password
    networks:
      - tech-challenge-network-local

  redis:
    container_name: redis
    image: library/redis:6-alpine
    platform: linux/amd64
    ports:
      - "6380:6379"
    healthcheck:
      test: [ "CMD", "redis-cli", "-p", "6379", "ping" ]
      interval: 5s
      timeout: 5s
      retries: 3
    networks:
      - tech-challenge-network-local

  cdn:
    container_name: cdn
    image: nginx:alpine
    platform: linux/amd64
    restart: always
    ports:
      - "8888:8080"
    volumes:
      - ./docker/nginx/nginx.conf:/etc/nginx/nginx.conf
      - ./docker/nginx/static:/tech-challenge/static
    networks:
      - tech-challenge-network-local

networks:
  tech-challenge-network-local:
    driver: bridge

